### PR TITLE
Added sihas energy monitor for wwst

### PR DESCRIPTION
**Pull request by shinasys**
model: "PMM-300Z2", "PMM-300Z3"

### CODE REVIEW

**Comment by greens** [comment]
luacheck:

```
    drivers/SmartThings/zigbee-power-meter/src/shinasystems/init.lua:80:9: (W311) value assigned to variable delta_tick is overwritten on line 84 before use
```

Also, I'd like to see the tests for this device expanded to cover the new functionality. And I'm a bit concerned that this line:
`device:set_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY, 1, {persist = true})` changing didn't break any tests.

**Comment by greens** [comment]
Also, as indicated, your new profile is not necessary. Please use the existing one.

**Comment by shinasys** [comment]
> Also, as indicated, your new profile is not necessary. Please use the existing one.

The metadata of the 'power-meter-consumption-report.yml' file has the name 'EZEX' in the 'vid' field, so I added a new one. If it's okay, I will use 'power-meter-consumption-report.yml'.

**Comment by greens** [comment]
@shinasys please add some tests for this new behavior to your pull request

**Comment by greens** [comment]
Looks like your tests are failing:

```
zigbee-power-meter: 
	test zigbee power meter consumption report sihas FAILED on SimpleMetering event should be handled by powerConsumptionReport capability
	test zigbee power meter consumption report sihas FAILED on lifecycle configure event should configure device
```

You can find info on setting them up to run locally here:

https://developer.smartthings.com/docs/devices/hub-connected/set-up-dev-env

and here:

https://developer.smartthings.com/docs/devices/hub-connected/test-your-driver

**Review by greens — APPROVED** [review]
the tests passed, that failure mark is an issue with the CI setup that I'm working on

**Comment by greens** [comment]
@shinasys sorry, it's not immediately clear to me what's wrong in the screenshot. Can you explain more?

**Comment by shinasys** [comment]
@greensIn the graph, we can see two peak values. The sum of these two peak values is greater than the total accumulated value. The two peak values should be displayed as several KWh to tens of KWh, which is the actual usage of this device, instead of 5.9 MWh.

**Comment by greens** [comment]
@shinasys It's still not really clear to me, but perhaps it might help you to track down the issue by writing some tests that verify the `powerConsumption` events that the driver creates. I noticed there isn't any coverage of that code in your tests.

**Comment by shinasys** [comment]
@greens The issue being reported is not that the powerConsumption is strange, but rather that the graph of the energyMeter is peculiar.
Anyway, I will take some more time and check this part again. If the problem persists, I will report it once more.

File: drivers/SmartThings/zigbee-power-meter/src/shinasystems/init.lua [hunk 1]
```diff
@@ -74,8 +74,7 @@ local function energy_meter_handler(driver, device, value, zb_rx)
     device:set_field(constants.ENERGY_METER_OFFSET, offset, {persist = true})
   end
   raw_value_kilowatts = raw_value_kilowatts - offset
-  raw_value_kilowatts = math.floor((raw_value_kilowatts*10^3+0.5))/(10^3) -- Let's round it to 3 decimal(W) places.
```
Comment (shinasys) [inline drivers/SmartThings/zigbee-power-meter/src/shinasystems/init.lua]: By removing this line (math.floor) and testing for four days, the strange graphs no longer appear.

**Comment by greens** [comment]
>    drivers/SmartThings/zigbee-power-meter/src/shinasystems/init.lua:77:1: (W611) line contains only whitespace

**Comment by greens** [comment]
>    drivers/SmartThings/zigbee-power-meter/src/test/test_zigbee_power_meter_consumption_report_sihas.lua:86:49: (W113) accessing undefined variable curr_save_tick
    drivers/SmartThings/zigbee-power-meter/src/test/test_zigbee_power_meter_consumption_report_sihas.lua:106:49: (W113) accessing undefined variable curr_save_tick

**Pull request merged**